TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/cluster.py

```diff
@@ -103,6 +110,17 @@ def get_basic_auth_password(self) -> str:
         password = client.get_secret_value(SecretId=self.auth_details["password_from_secret_arn"])
         return password["SecretString"]
 
+    def _get_sigv4_details(self, force_region=False) -> tuple[str, str]:
+        """Return the service signing name and region name. If force_region is true,
+        it will instantiate a boto3 session to guarantee that the region is not None.
+        This will fail if AWS credentials are not available.
+        """
+        assert self.auth_type == AuthMethod.SIGV4
+        if force_region:
+            session = boto3.session.Session()
+            return self.auth_details.get("service", "es"), self.auth_details.get("region", session.region_name)
```

Comment:
nit: Can we initiate the boto3 session in such a way that it's only initiated if region is None?